VueJS/components/ListItem.js

```diff
@@ -1 +1,32 @@
-export default Vue.component('list-item', {})
+export default Vue.component('list-item', {
+
+    template: `<li>
+	            <template v-if="edit">
+		            {{ item.id }} {{ item.message }}
+		            <button v-on:click="edit=false">Edit</button>
+	            </template>
+	            <template v-else>
+		            <input placeholder="Add Todo here" v-model="input"/>
+		            <button v-on:click="save">Save</button>\
+		            <button v-on:click="edit=true">Cancel</button>
+	            </template>
+	            <button v-on:click="remove">Remove</button>
+            </li>`,
+    data: function(){
+        return{
+            input: this.item.message,
+            edit: !(this.item.message ===''),
```

Comment:
This is very obscure and hard to read. I know what you want to do because I saw it in so many other assignments.

You can only understand it if you read the code in another location where you create a new todo with an empty `message`. Then you want to have the input field visible. But what if you want to initialize the todo with a non-empty message?

Actually it's bad user experience to save the todo to the list before you have the chance to edit it. So, put a form next to the "Add Todo" button where you can enter the message of the new todo. If you do that you are free to simply assign a `isEditing` or `editMode` to `false` in this very location.